examples/apps/ai-collab/src/components/UserPresenceGroup.tsx

```diff
@@ -22,39 +22,28 @@ interface UserPresenceProps {
 const UserPresenceGroup: React.FC<UserPresenceProps> = ({
 	userPresenceGroup,
 }): JSX.Element => {
+	const isFirstRender = useRef(true);
 	const [photoUrls, setPhotoUrls] = useState<string[]>([]);
-	const [isPhotoFetched, setIsPhotoFetched] = useState<boolean>(false);
 
-	// this effect will run when the userPresenceGroup changes and will update the photoUrls state
-	useEffect(() => {
+	const fetchAllPhotos = (): string[] => {
 		const allPhotos: string[] = [];
-		for (const element of [...userPresenceGroup.onlineUsers.clientValues()]) {
-			for (const user of [...element.items.values()]) {
-				allPhotos.push(user.value.value.photo);
-			}
-		}
-		setPhotoUrls(allPhotos);
-	}, [userPresenceGroup]);
+		// eslint-disable-next-line unicorn/no-array-for-each
+		userPresenceGroup.props.onlineUsers.local.forEach((user) => {
+			allPhotos.push(user.value.photo);
+		});
+		return allPhotos;
+	};
```

Comment:
My first comment was going to be "no need to go against the linter, use a regular `for .. of` loop", but we can do even better: the whole function can be transformed into `return userPresenceGroup.props.onlineUsers.local.map((user) => user.value.photo)`, and since we only seem to have one call site for it, I'd suggest just inlining that `map()` call in there. The function is currently also used as a dependency of a React effect below but I don't think that's correct? I thought only React state needs to be specified as dependencies on effects.

Reply:
But the `forEach()` function here is provided by `LatestMapValueManager`. I think we should use the self-defined function of the object.

Reply:
This is an interesting question. @jason-ha is there an advantage to using .forEach off of a ValueMap as opposed to a for .. of ?